broker/provision_endpoint.go

```diff
@@ -119,3 +119,62 @@ func (bkr *Broker) assertProvisionPrecondition(instanceID structs.ClusterID, fea
 
 	return bkr.scheduler.VerifyClusterFeatures(features)
 }
+
```

Comment:
Currently these 3 helpers aren't used outside of `provision_endpoint.go` so I'll leave them here. If they get some reuse then yeah that sounds good.